contracts/examples/mintable-erc721-transfer/FxMintableERC721ChildTunnel.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {FxBaseChildTunnel} from "../../tunnel/FxBaseChildTunnel.sol";
+import {Create2} from "../../lib/Create2.sol";
+import {FxMintableERC721} from "../../tokens/FxMintableERC721.sol";
+import {IERC721Receiver} from "../../lib/IERC721Receiver.sol";
+import {Ownable} from "../../lib/Ownable.sol";
+import {Address} from "../../lib/Address.sol";
+
+/**
+ * @title FxMintableERC721ChildTunnel
+ */
+contract FxMintableERC721ChildTunnel is FxBaseChildTunnel, Create2, IERC721Receiver, Ownable {
+    bytes32 public constant DEPOSIT = keccak256("DEPOSIT");
+    // bytes32 public constant MAP_TOKEN = keccak256("MAP_TOKEN");
+    string public constant SUFFIX_NAME = " (FXERC721)";
+    string public constant PREFIX_SYMBOL = "fx";
+
+    event TokenMapped(address indexed rootToken, address indexed childToken);
+    event FxWithdrawMintableERC721(
+        address indexed rootToken,
+        address indexed childToken,
+        address indexed userAddress,
+        uint256 id
+    );
+    event FxDepositMintableERC721(
+        address indexed rootToken,
+        address indexed depositor,
+        address indexed userAddress,
+        uint256 id
+    ); // root to child token
+    mapping(address => address) public rootToChildToken;
+    // child token template
+    address public immutable childTokenTemplate;
+    // root token template codehash
+    bytes32 public immutable rootTokenTemplateCodeHash;
+
+    constructor(
+        address _fxChild,
+        address _childTokenTemplate,
+        address _rootTokenTemplate
+    ) FxBaseChildTunnel(_fxChild) {
+        childTokenTemplate = _childTokenTemplate;
+        require(Address.isContract(_childTokenTemplate), "FxMintableERC721ChildTunnel: Token template is not contract");
```

Comment:
This can be moved one line above to achieve the same effect and save some gas in revert cases.

Reply:
2f54fd4c86643b4993c5022a756662744b436a8f